src/components/SearchInput.tsx

```diff
@@ -1,6 +1,31 @@
 import { SearchIcon } from "./Icons";
+import { useState, useCallback } from "react";
+import { useSearchParams, useNavigate } from "react-router-dom";
 
 const SearchInput = () => {
+  const navigate = useNavigate();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [searchValue, setSearchValue] = useState(searchParams.get("q") || "");
+
+  const debouncedSearch = useCallback(
+    debounce((query: string) => {
+      if (query) {
+        setSearchParams({ q: query });
+        navigate(`/search?q=${encodeURIComponent(query.trim().toLowerCase())}`);
```

Comment:
I don't think navigating urls every debounced request is a good thing. I know it is debounced but still sometimes people pause to think while searching, and it will clutter the history.

You can maybe search on `Enter`, or show results while typing but navigate the URL on Enter or onBlur

Reply:
It's an SPA so it doesn't "refresh" the page, but yeah it will clutter the history, what you can do is use the replace feature of navigate [here](https://api.reactrouter.com/v7/interfaces/react_router.NavigateOptions.html#replace)

Reply:
I think on Enter and onBlur is a great idea! I'll look into it.